File: src/editor/filter/leaf/options/single/LayoutFilter.java
package editor.filter.leaf.options.single;

import editor.database.card.Card;
import editor.database.card.CardLayout;
import editor.filter.Filter;
import editor.filter.FilterAttribute;

import java.util.HashSet;

/**
 * This class represents a filter that groups cards by layout.
 *
 * @author Alec Roelke
 */
public class LayoutFilter extends SingletonOptionsFilter<CardLayout>
{
    /**
     * Create a new LayoutFilter.
     */
    public LayoutFilter()
    {
        super(FilterAttribute.LAYOUT, Card::layout);
    }

    @Override
    public CardLayout convertFromString(String str)
    {
        return CardLayout.valueOf(str.replace(' ', '_').toUpperCase());
    }

    @Override
    public Filter copy()
    {
        LayoutFilter filter = (LayoutFilter)FilterAttribute.createFilter(FilterAttribute.LAYOUT);
        filter.contain = contain;
        filter.selected = new HashSet<>(selected);
        return filter;
    }
}

File: src/main/scala/editor/filter/leaf/options/single/LayoutFilter.scala
package editor.filter.leaf.options.single

import editor.database.attributes.CardAttribute
import editor.database.card.Card
import editor.database.card.CardLayout
import editor.filter.FaceSearchOptions
import editor.util.Containment

/**
 * A filter that groups cards based on layout.
 * @author Alec Roelke
 */
final case class LayoutFilter(contain: Containment = Containment.AnyOf, selected: Set[CardLayout] = Set.empty) extends SingletonOptionsFilter[CardLayout, LayoutFilter] {
  override def faces = FaceSearchOptions.ANY
  override def attribute = CardAttribute.Layout
  override val unified = true
  override def value = _.layout
  override def copy(faces: FaceSearchOptions, contain: Containment, selected: Set[CardLayout]) = copy(contain = contain, selected = selected)
  override def copyFaces(faces: FaceSearchOptions) = this
}